crates/noirc_evaluator/src/ssa/context.rs

```diff
@@ -936,10 +936,26 @@ impl SsaContext {
     fn init_array(&mut self, array_id: ArrayId, stack_frame: &mut StackFrame) {
         let len = self.mem[array_id].len;
         let e_type = self.mem[array_id].element_type;
-        for i in 0..len {
+        let mut values = Vec::new();
+        for _ in 0..len {
+            values.push(self.zero_with_type(e_type));
+        }
```

Comment:
Apologies, I seem to have missed this on the last review. This should be `let values = vecmap(0..len, |_| self.push_zero_with_type(e_type))`

Reply:
Done, in fact we don't even need the vecmap.

Reply:
Right, you could probably `vec![self.zero_with_type(e_type); len]`

Reply:
Oh thats what you've done; never mind